src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs

```diff
@@ -433,11 +435,9 @@ private bool PerformCompilation()
                 {
                     projectCompilationChanged = UpdateProjectCompilation(project, out compilation);
 
-                    project.Diagnostics = new DiagnosticsMessage
-                    {
-                        Framework = project.Sources.Framework,
-                        Diagnostics = compilation.Diagnostics
-                    };
+                    project.Diagnostics = new DiagnosticsMessage(
+                        compilation.Diagnostics,
+                        project.Sources.Framework);
```

Comment:
The `DiagnosticsMessage` now also carries diagnostics information other than _compilation diagnostics_. The original interface doesn't distinguish two initialization approaches well enough.